clang-tools-extra/clangd/index/SymbolCollector.cpp

```diff
@@ -576,6 +578,27 @@ SymbolCollector::getRefContainer(const Decl *Enclosing,
   return Enclosing;
 }
 
+std::vector<CXXConstructorDecl *>
+SymbolCollector::findIndirectConstructors(const Decl *D) {
+  auto *FD = llvm::dyn_cast<clang::FunctionDecl>(D);
+  if (FD == nullptr || !FD->isTemplateInstantiation())
+    return {};
+  if (auto *PT = FD->getPrimaryTemplate();
+      PT == nullptr || !isLikelyForwardingFunction(PT))
+    return {};
+  if (auto Entry = ForwardingToConstructorCache.find(FD);
```

Comment:
A hashtable lookup is going to be quicker than an `isLikelyForwardingFunction` check, let's switch the order?